modules/gdscript/gdscript_compiler.cpp

```diff
@@ -2968,7 +2973,7 @@ Error GDScriptCompiler::compile(const GDScriptParser *p_parser, GDScript *p_scri
 		GDScriptCache::add_static_script(p_script);
 	}
 
-	return GDScriptCache::finish_compiling(main_script->get_path());
+	return GDScriptCache::finish_compiling(main_script->path);
```

Comment:
This was the issue with testing, which doesn't use `ResourceLoader::load()` to load scripts, meaning that `Resource::path`, which you get using `main_script->get_path()` was empty, so that `GDScriptCache::finish_compiling()` was getting an empty string and being a no-op :)